internal/resources/gitrepository/resource_git_repository.go

```diff
@@ -40,3 +55,259 @@ var gitRepositorySchema = map[string]*schema.Schema{
+	config := m.(authctx.TanzuContext)
+
+	gitRepositoryName, ok := d.Get(nameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read git repository name")
+	}
+
+	gitRepositoryNamespaceName, ok := d.Get(namespaceNameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read git repository namespace name")
+	}
+
+	scopedFullnameData := scope.ConstructScope(d, gitRepositoryName, gitRepositoryNamespaceName)
+
+	if scopedFullnameData == nil {
+		return diag.Errorf("Unable to delete Tanzu Mission Control git repository entry; Scope full name is empty")
+	}
+
+	err := enableContinuousDelivery(&config, scopedFullnameData, common.ConstructMeta(d))
+	if err != nil {
+		return diag.FromErr(errors.Wrapf(err, "Unable to delete Tanzu Mission Control git repository entry, name : %s", gitRepositoryName))
+	}
+
+	switch scopedFullnameData.Scope {
+	case commonscope.ClusterScope:
+		if scopedFullnameData.FullnameCluster != nil {
+			err := config.TMCConnection.ClusterGitRepositoryResourceService.VmwareTanzuManageV1alpha1ClusterFluxcdGitrepositoryResourceServiceDelete(scopedFullnameData.FullnameCluster)
+			if err != nil && !clienterrors.IsNotFoundError(err) {
+				return diag.FromErr(errors.Wrapf(err, "Unable to delete Tanzu Mission Control cluster git repository entry, name : %s", gitRepositoryName))
+			}
+		}
+	case commonscope.ClusterGroupScope:
+		if scopedFullnameData.FullnameClusterGroup != nil {
+			err := config.TMCConnection.ClusterGroupGitRepositoryResourceService.VmwareTanzuManageV1alpha1ClustergroupFluxcdGitrepositoryResourceServiceDelete(scopedFullnameData.FullnameClusterGroup)
+			if err != nil && !clienterrors.IsNotFoundError(err) {
+				return diag.FromErr(errors.Wrapf(err, "Unable to delete Tanzu Mission Control cluster group git repository entry, name : %s", gitRepositoryName))
+			}
+		}
+	case commonscope.UnknownScope:
+		return diag.Errorf("no valid scope type block found: minimum one valid scope type block is required among: %v. Please check the schema.", strings.Join(scope.ScopesAllowed[:], `, `))
+	}
+
+	// d.SetId("") is automatically called assuming delete returns no errors, but
+	// it is added here for explicitness.
+	d.SetId("")
+
+	return diags
+}
+
+func resourceGitRepositoryInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	gitRepositoryName, ok := d.Get(nameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read git repository name")
+	}
+
+	gitRepositoryNamespaceName, ok := d.Get(namespaceNameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read git repository namespace name")
+	}
+
+	scopedFullnameData := scope.ConstructScope(d, gitRepositoryName, gitRepositoryNamespaceName)
+
+	if scopedFullnameData == nil {
+		return diag.Errorf("Unable to update Tanzu Mission Control git repository entry; Scope full name is empty")
+	}
+
+	err := enableContinuousDelivery(&config, scopedFullnameData, common.ConstructMeta(d))
```

Comment:
This is wrong right, 
In the case of update, if the continuous delivery is disabled, the call should be create and not update

Reply:
As discussed we will be enabling the feature only in case of `create`

The `update` which internally calls a `read` before checking for changes will not enable the feature if an org admin has disabled it.

Reply:
Updated.